src/apps/jsgeneric/jsgeneric.cpp

```diff
@@ -61,63 +133,104 @@ namespace ccfapp
           return;
         }
 
-        nlohmann::json response = {};
-        /*
-        const auto response = tsr->run<nlohmann::json>(
-          args.tx,
-          {*handler_script,
-           {},
-           WlIds::USER_APP_CAN_READ_ONLY,
-           scripts->get(UserScriptIds::ENV_HANDLER)},
-          // vvv arguments to the script vvv
-          args);
-        */
-
-        auto err_it = response.find("error");
-        if (err_it == response.end())
+        JSRuntime* rt = JS_NewRuntime();
+        if (rt == nullptr)
         {
-          auto result_it = response.find("result");
-          if (result_it == response.end())
-          {
-            // Response contains neither result nor error. It may not even be an
-            // object. We assume the entire response is a successful result.
-            args.rpc_ctx->set_response_result(std::move(response));
-            return;
-          }
-          else
-          {
-            args.rpc_ctx->set_response_result(std::move(*result_it));
-            return;
-          }
+          throw std::runtime_error("Failed to initialise QuickJS runtime");
         }
-        else
+        // TODO: share runtime across handlers?
+        // TODO: set memory limit with JS_SetMemoryLimit
+
+        JSContext* ctx = JS_NewContext(rt);
+        if (ctx == nullptr)
         {
-          int err_code = jsonrpc::CCFErrorCodes::SCRIPT_ERROR;
-          std::string msg = "";
+          JS_FreeRuntime(rt);
+          throw std::runtime_error("Failed to initialise QuickJS context");
```

Comment:
Should be "Failed to free".

Reply:
No? A parent runtime can have multiple contexts (each one has its own heap), and here we're freeing the (sole) parent runtime if we fail to create a context successfully, but the error is that we failed to initialise a context.